moduleframework/tools/modulelint.py

```diff
@@ -43,57 +43,75 @@ def setUp(self):
         if self.dp.dockerfile is None:
             self.skip()
 
-    def testArchitectureInEnvAndLabelExists(self):
+    def test_architecture_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("ARCH="))
         self.assertTrue(self.dp.get_specific_label("architecture"))
 
-    def testNameInEnvAndLabelExists(self):
+    def test_name_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("NAME="))
         self.assertTrue(self.dp.get_specific_label("name"))
 
-    def testReleaseLabelExists(self):
+    def test_release_label_xists(self):
         self.assertTrue(self.dp.get_specific_label("release"))
 
-    def testVersionLabelExists(self):
+    def test_version_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("version"))
 
-    def testComRedHatComponentLabelExists(self):
+    def test_com_redhat_component_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("com.redhat.component"))
 
-    def testIok8sDescriptionExists(self):
-        self.assertTrue(self.dp.get_specific_label("io.k8s.description"))
 
-    def testIoOpenshiftExposeServicesExists(self):
-        label_io_openshift = "io.openshift.expose-services"
-        exposes = self.dp.get_docker_expose()
-        label_list = self.dp.get_docker_labels()
-        self.assertTrue(label_list[label_io_openshift])
-        for exp in exposes:
-            self.assertTrue("%s" % exp in label_list[label_io_openshift])
+class DockerfileLinterInContainer(container_avocado_test.ContainerAvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    def test_docker_nodocs(self):
+        self.start()
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", verbose=False).stdout
+        # This returns a list of packages defined in config.yaml for testing
+        # e.g. ["bash", "rpm", "memcached"] in case of memcached
+        pkgs = self.backend.getPackageList()
+        list_pkg = [pkg for pkg in installed_pkgs.split('\n') if pkg in pkgs]
+        for pkg in list_pkg:
+            all_docs = self.run("rpm -qd %s" % pkg, verbose=False).stdout
+            for doc in all_docs.strip().split('\n'):
+                self.assertNotEqual(0, self.run("test -e %s" % doc, ignore_status=True).exit_status)
+
+    def test_docker_clean_all(self):
+        self.start()
+        pkg_mgr = "yum"
+        # Detect distro in image
+        distro = self.run("cat /etc/os-release").stdout
 
-    def testIoOpenShiftTagsExists(self):
-        label_list = self.dp.get_docker_labels()
-        self.assertTrue("io.openshift.tags" in label_list)
+        if 'NAME=Fedora' in distro:
+            pkg_mgr = "dnf"
+        # Look, whether we have solv files in /var/cache/<pkg_mgr>/*.solv
+        # dnf|yum clean all deletes the file *.solv
+        ret = self.run("ls /var/cache/%s/*.solv" % pkg_mgr, ignore_status=True)
```

Comment:
doesn't work on RHEL, there are no `.solv` files in there:

```
[root@b06253698a54 /]# yum makecache
Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
extras               | 3.0 kB  00:00:00
rhel-7-server-rpms   | 4.1 kB  00:00:00
Metadata Cache Created

[root@b06253698a54 /]# cd /var/cache/yum

[root@b06253698a54 yum]# ll
total 0
drwxr-xr-x. 3 root root 21 Oct  4 11:34 x86_64

[root@b06253698a54 yum]# ll x86_64/7Server/
total 16
drwxr-xr-x. 4 root root 4096 Oct  9 09:17 extras
drwxr-xr-x. 4 root root 4096 Oct  9 09:17 rhel-7-server-rpms
-rw-r--r--. 1 root root   46 Oct  9 09:17 timedhosts

[root@b06253698a54 yum]# du -sh .
106M    .

[root@b06253698a54 yum]# yum clean all
Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
Cleaning repos: extras rhel-7-server-rpms
Cleaning up everything
Maybe you want: rm -rf /var/cache/yum, to also free up space taken by orphaned data from disabled or removed repos

[root@b06253698a54 yum]# du -sh .
8.0K    .
```

I would probably check via `du` whether `/var/cache/yum` exceeds certain size, something like `128K`.